controller/src/test/java/org/jboss/as/controller/test/GlobalOperationsTestCase.java

```diff
@@ -954,6 +955,20 @@ public void testReadResourceAttributesOnly() throws Exception {
         executeForFailure(operation);
     }
 
+    @Test
+    public void testRecursiveReadSubModeWithResolveExpression() throws Exception {
+            try {   ModelNode operation = createOperation(READ_RESOURCE_OPERATION, "profile", "profileA");
+            System.setProperty("${expr}", "test-resolve-true");
+            operation.get(RECURSIVE).set(true);
+            operation.get(RESOLVE_EXPRESSIONS).set(true);
+            ModelNode result = executeForResult(operation);
+            result.get("subsystem", "subsystem2", "expression").set(System.getProperty("${expr}"));
```

Comment:
this line must be removed: you are setting the value of the `expression` attribute that will override any value returned by WildFly. This makes the test pass without any confirmation that the expression has been resolved recursively by WildFly

Reply:
Done,please check,I missed removing those lines earlier as was testing it again and again.